src/js/view/View.js

```diff
@@ -0,0 +1,164 @@
+    this.sendSaveRequest = sendSaveRequest;
+  }
+
+  #openModal = () => {
+    this.modalContainer.classList.remove('hide');
+  };
+
+  #handleSearch = async (event) => {
+    event.preventDefault();
+    this.#clearNoResult();
+    scrollToTop(this.videoList);
+    const { value: keyword } = this.searchInputKeyword;
+    if (isBlankValue(keyword)) {
+      return;
+    }
+    removeElementList([...this.videoList.childNodes]);
+    this.#loadSkeleton();
+    const searchResultArray = await this.sendSearchRequest(keyword);
+    this.#renderSearchResult(searchResultArray);
+  };
+
+  #handleScrollToLastItem() {
+    return new IntersectionObserver(
+      async (entries) => {
+        if (entries[0].isIntersecting) {
+          this.requestMoreResult.unobserve(this.videoList.lastChild);
+          this.#loadSkeleton();
+          const moreResult = await this.sendLoadMoreRequest();
+          this.#renderSearchResult(moreResult);
+        }
+      },
+      { threshold: 0.5 }
+    );
+  }
+
+  #handleVideoSaveClick = (event) => {
+    try {
+      this.sendSaveRequest(event.target.dataset.videoId);
+      event.target.disabled = true;
+    } catch (error) {
+      alert(error.message);
+    }
+  };
+
+  #renderSearchResult(searchResultArray) {
+    const skeletonList = this.videoList.querySelectorAll('.skeleton');
+    removeElementList(skeletonList);
+
+    if (this.#isEndOfResult(searchResultArray)) return;
+    if (this.#isNoResult(searchResultArray)) {
+      this.#renderNoResult();
+      return;
+    }
+
+    const resultElementArray = this.#createElementFromObject(searchResultArray);
+    this.videoList.append(...resultElementArray);
+    this.requestMoreResult.observe(this.videoList.lastChild);
+  }
+
+  #isEndOfResult(searchResultArray) {
+    return searchResultArray === null;
+  }
+
+  #isNoResult(searchResultArray) {
+    return searchResultArray.length === 0;
+  }
+
+  #createElementFromObject(searchResultArray) {
+    return searchResultArray.map((resultItem) => this.#createVideoElement(resultItem));
+  }
+
+  #createVideoElement(resultItem) {
+    const videoElement = document.createElement('li');
+    videoElement.className = 'video-item';
+    videoElement.insertAdjacentHTML('beforeend', this.#videoElementTemplate(resultItem));
+    selectDom('.video-item__save-button', videoElement).addEventListener(
+      'click',
+      this.#handleVideoSaveClick
+    );
+    return videoElement;
+  }
+
+  #videoElementTemplate({ thumbnail, title, channelTitle, publishedAt, videoId, isSaved }) {
+    return `<img src="${thumbnail}" alt="video-item-thumbnail" class="video-item__thumbnail">
+    <h4 class="video-item__title">${title}</h4>
+    <p class="video-item__channel-name">${channelTitle}</p>
+    <p class="video-item__published-date">${formatDateString(publishedAt)}</p>
+    <button 
+      ${isSaved && 'disabled'}
+      class="video-item__save-button button"
+      data-video-id="${videoId}"
+    >
+      ⬇ 저장
+    </button>`;
+  }
```

Comment:
개인적인 취향인 문제인데 인덴트 맞춰주시는게 코드가 눈에 더 잘 들어오는 것 같아요!
```suggestion
    return`
      <img src="${thumbnail}" alt="video-item-thumbnail" class="video-item__thumbnail">
      <h4 class="video-item__title">${title}</h4>
      <p class="video-item__channel-name">${channelTitle}</p>
      <p class="video-item__published-date">${formatDateString(publishedAt)}</p>
      <button 
        ${isSaved && 'disabled'}
        class="video-item__save-button button"
        data-video-id="${videoId}"
      >
        ⬇ 저장
      </button>
    `;}
```

Reply:
그렇네요! 인덴트를 맞춰서 html 의 위계를 좀 더 쉽게 파악할 수 있도록 하는 것이 더 보기 좋을 것 같습니다!